src/test/rpc_wallet_tests.cpp

```diff
@@ -404,10 +404,8 @@ BOOST_AUTO_TEST_CASE(rpc_wallet_z_exportwallet)
     BOOST_CHECK(addrs.size()==0);
 
     // wallet should have one key
-    auto address = pwalletMain->GenerateNewZKey();
-    BOOST_CHECK(IsValidPaymentAddress(address));
-    BOOST_ASSERT(boost::get<libzcash::SproutPaymentAddress>(&address) != nullptr);
-    auto addr = boost::get<libzcash::SproutPaymentAddress>(address);
+    auto addr = pwalletMain->GenerateNewSproutZKey();
+    BOOST_CHECK(IsValidPaymentAddress(addr));
```

Comment:
Wondering if it's worth defining a `IsValidSproutAddress` for here

Reply:
It looks like `IsValidPaymentAddress` returns true iff the type is not `InvalidEncoding` which it can't be if we have `SproutPaymentAddresses`.  I don't think we need to add this method, and in fact I think I will remove this check here.

Typically we are calling `IsValidPaymentAddress` immediately after having `DecodePaymentAddress` for some string.. In these cases we do not know the type.